### PR TITLE
helvum: 0.3.1 -> 0.3.2

### PR DESCRIPTION
**Pull request by fufexan**
###### Motivation for this change

Switched to meson build system, which also installs a .desktop file and icons. Took inspiration from #116324 when rewriting the derivation.
Supersedes #150081.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 150157` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>helvum</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
Package is missing a `meta.platforms`, prolly `platforms.linux`?

**Review by jansol — APPROVED** [review]
Nothing jumps out as obviously problematic, although I'm a bit puzzled as to why it needs both gtk3 and gtk4?

Builds & works fine as far as I can tell.

**Comment by fufexan** [comment]
@jansol it needs `gtk3` for `gtk-update-icon-cache`. I couldn't find it in the `gtk4` package.

**Comment by bobby285271** [comment]
> @jansol it needs `gtk3` for `gtk-update-icon-cache`. I couldn't find it in the `gtk4` package.

I think we can use `gtk4-update-icon-cache`, maybe open an issue or PR upstream?

**Comment by fufexan** [comment]
@bobby285271 I opened a [MR](https://gitlab.freedesktop.org/ryuukyu/helvum/-/merge_requests/24).

**Comment by fufexan** [comment]
@bobby285271 can this be merged now?

File: pkgs/applications/audio/helvum/default.nix [hunk 1]
```diff
@@ -1,47 +1,73 @@
...
-  };
+  patches = [
+    # enables us to use gtk4-update-icon-cache instead of gtk3 one
+    (fetchpatch {
+      url = "https://gitlab.freedesktop.org/ryuukyu/helvum/-/merge_requests/24.patch";
+      sha256 = "sha256-WmI6taBL/6t587j06n0mwByQ8x0eUA5ECvGNjg2/vtk=";
+    })
+  ];
+
+  prePatch = ''
```
Comment (bobby285271) [inline pkgs/applications/audio/helvum/default.nix:62]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/audio/helvum/default.nix [hunk 2]
```diff
@@ -1,47 +1,73 @@
...
+    patchShebangs build-aux/cargo.sh
+  '';
 
   meta = with lib; {
     description = "A GTK patchbay for pipewire";
     homepage = "https://gitlab.freedesktop.org/ryuukyu/helvum";
     license = licenses.gpl3Only;
     maintainers = with maintainers; [ fufexan ];
+    platforms = lib.platforms.linux;
```
Comment (bobby285271) [inline pkgs/applications/audio/helvum/default.nix:71]:
```suggestion
    platforms = platforms.linux;
```

**Pull request merged**